### PR TITLE
Subscription change email alerts

### PR DESCRIPTION
**Pull request by proteusvacuum**
http://manage.dimagi.com/default.asp?174032#969804
http://manage.dimagi.com/default.asp?174037#969805

This was a hairy one... The email gets sent in `apply_upgrades_and_downgrades`, which gets called from both `new_domain_subscription` _and_ `change_plan`. The properties were getting set _after_ the email had been set, which is why it was good in the admin interface but wrong in the email. 

Probably a good candidate for some refactoring at a later point.

@nickpell 
code buddy: @biyeun

### CODE REVIEW

File: corehq/apps/accounting/models.py [hunk 1]
```diff
@@ -985,7 +985,9 @@ def update_subscription(self, date_start=None, date_end=None,
     def change_plan(self, new_plan_version, date_end=None,
                     note=None, web_user=None, adjustment_method=None,
                     service_type=None, pro_bono_status=None,
-                    transfer_credits=True, internal_change=False):
+                    transfer_credits=True, internal_change=False, account=None,
+                    do_not_invoice=None, **kwargs
+    ):
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/domain/forms.py [hunk 2]
```diff
@@ -1558,14 +1564,14 @@ def process_subscription_management(self):
...
+                    service_type=SubscriptionType.CONTRACTED
                 )
-                new_subscription.account = self.next_account
-            if new_subscription.date_start <= datetime.date.today() and datetime.date.today() < new_subscription.date_end:
-                new_subscription.is_active = True
-            new_subscription.do_not_invoice = False
-            new_subscription.auto_generate_credits = True
-            new_subscription.service_type = SubscriptionType.CONTRACTED
-            new_subscription.save()
+                if new_subscription.date_start <= datetime.date.today() and datetime.date.today() < new_subscription.date_end:
```
Comment (dimagimon) [inline corehq/apps/domain/forms.py]: E501 line too long (126 > 115 characters)

File: corehq/apps/domain/forms.py [hunk 3]
```diff
@@ -1558,14 +1564,15 @@ def process_subscription_management(self):
...
                 )
-                new_subscription.account = self.next_account
-            if new_subscription.date_start <= datetime.date.today() and datetime.date.today() < new_subscription.date_end:
-                new_subscription.is_active = True
-            new_subscription.do_not_invoice = False
-            new_subscription.auto_generate_credits = True
-            new_subscription.service_type = SubscriptionType.CONTRACTED
-            new_subscription.save()
+                if (new_subscription.date_start <= datetime.date.today()
+                    and datetime.date.today() < new_subscription.date_end):
```
Comment (dimagimon) [inline corehq/apps/domain/forms.py]: E129 visually indented line with same indent as next logical line

**Comment by nickpell** [comment]
:+1:

**Pull request merged**